Fix spectator view custom profile fields

laurynmm opened the pull request:
Updates the server to send an empty list of `custom_profile_fields` in `page_params` for spectators, rather than not sending the field at all. Fixes an error on the frontend where `custom_profile_fields` was undefined when users tried to access a message's sender info.

Also, updates `render_user_info_popover` to load the `login_to_access modal` in spectator views, and updates the hotkey implementation for viewing a message's sender info, `show_sender_info`, to also load the `login_to_access modal`.

---

**Notes**:
- I added the second commit because I noticed with the server change, we now loaded a basic version of the user info popover for spectators. That basic version included the user's role and join date, as well as the three dot menu which opens up the modal for muting a user, see screenshots below. Note that if the spectator tries to mute the user, then the `login_to_access` modal appears. The help center article for the [public access option](https://zulip.com/help/public-access-option#information-that-can-be-accessed-via-api-when-web-public-streams-are-enabled) does say that information is available [via the API](https://zulip.com/help/public-access-option#information-about-users), but doesn't imply that it will be visible in the web-app. Currently, due to the bug, nothing is shown to the spectator when they click on a user's name/avatar or use the hotkey shortcut.
- I put the check for loading the `login_to_access` modal in `render_user_info_popover`. Alternatively, we could put that check in the three places (`show_user_info_popover_for_message`, `show_user_info_popover`, `.on("click", ".user-list-sidebar-menu-icon"`) that the render function is called. Since the user list isn't displayed in the spectator view, we could probably only put the check in the first two function calls. I went with the current implementation because seemed clearer that we didn't want that popover rendered in spectator views.
- I noted a bit of a delay and image blurriness when the user info popover is loaded in the dev environment for a logged-in user. This seems to be present when I manually test the current `main` without these changes, so I think this is an existing issue and not introduced by these changes.
- If we want the frontend changes, then these two commits can be squashed, but I didn't want to assume that's the behavior we want, so I kept them separate for ease of reviewing and making updates/changes.

---

**Screenshots and screen captures:**

<details>
<summary>Current behavior in dev environment</summary>

![Screenshot from 2022-10-13 15-15-11](https://user-images.githubusercontent.com/63245456/195606609-ba767735-4fc1-4f45-b968-c017bd46d984.png)
</details>


<details>
<summary>After the server sends the empty list for custom profile fields</summary>

![Screenshot from 2022-10-13 13-42-42](https://user-images.githubusercontent.com/63245456/195606749-a29b1d44-1c9d-41bb-8ddd-d1313c5f877b.png)
![Screenshot from 2022-10-13 15-21-39](https://user-images.githubusercontent.com/63245456/195608166-b36ea411-12ef-4da8-9ac5-4659fc18f14e.png)
![Screenshot from 2022-10-13 15-21-17](https://user-images.githubusercontent.com/63245456/195608160-520a8a7d-dc6e-4af4-a4a6-5a106528ba2a.png)
</details>


<details>
<summary>After updating the frontend for popovers and hotkeys</summary>

![Screenshot from 2022-10-13 14-01-12](https://user-images.githubusercontent.com/63245456/195606925-00f98c96-f262-4e23-8c31-93c02710aae3.png)
![Screenshot from 2022-10-13 14-01-37](https://user-images.githubusercontent.com/63245456/195606930-b704cfa9-b403-4c85-80fc-2676dd54115d.png)
</details>

---

**Self-review checklist**

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/version-control.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.

timabbott commented:
Hmm, I think we do want to be able to click on the avatars and get the larger avatar rather than just asking users to login; @alya thoughts?

We will need to instead tweak the logic further to not offer "Mute the user" if we want to do that.

alya commented:
Yes, that's right -- we should show the profile (without login), but hide the three-dot menu.

laurynmm commented:
**Updates**:
- Removed the commit that added the `login_to_access` modal when accessing a user's info popover in a spectator view.
- Updated so that manage user menu, which opens a popover for muting the user, isn't shown for spectator views.

**Screenshots**:

<details>
<summary>User info popover in spectator view</summary>

![Screenshot from 2022-10-14 11-29-33](https://user-images.githubusercontent.com/63245456/195820759-0a37038c-6f03-4726-8035-47a46d746f20.png)
![Screenshot from 2022-10-14 11-29-51](https://user-images.githubusercontent.com/63245456/195820770-9131f579-f47b-4d4f-9752-93eaf15dda62.png)
![Screenshot from 2022-10-14 11-29-59](https://user-images.githubusercontent.com/63245456/195820774-3c1880eb-aa23-43b1-8550-c3a11a5a7136.png)
</details>

alya commented:
Looks good to me in manual testing!

@timabbott tagging for your review.

timabbott commented:
Merged, thanks @laurynmm!
